demo/runtime/src/runtime/system.rs

```diff
@@ -176,22 +168,38 @@ fn final_checks(block: &Block) {
 	let ref header = block.header;
 
 	// check digest
-	with_env(|e| {
-		assert!(header.digest == e.digest);
-	});
+	assert!(header.digest == Digest::get());
+
+	// remove temporaries.
+	kill_temps();
 
 	// check storage root.
 	let storage_root = storage_root().into();
 	info_expect_equal_hash(&header.state_root, &storage_root);
 	assert!(header.state_root == storage_root, "Storage root must match that calculated.");
 }
 
+fn kill_temps() {
+	Number::kill();
+	ParentHash::kill();
+	RandomSeed::kill();
+	Digest::kill();
+	TransactionsRoot::kill();
+}
+
 fn post_finalise(header: &Header) {
 	// store the header hash in storage; we can't do it before otherwise there would be a
 	// cyclic dependency.
 	BlockHashAt::insert(&header.number, &header.blake2_256().into());
 }
 
+fn calculate_random() -> Hash {
+	let c = block_number() - 1;
+	(0..81)
+		.map(|i| if c >= i { block_hash(c - i) } else { Default::default() })
```

Comment:
so for the first 81 blocks the random seed won't be random at all

Reply:
it'll still be a bit random. but sure, lower security. thankfully it doesn't really matter.